cmake/modules/FindMySQL.cmake

```diff
@@ -10,11 +10,19 @@ if(MYSQL_INCLUDE_DIR OR MYSQL_)
 endif()
 
 if(NOT WIN32)
-  find_program(MYSQL_CONFIG_EXECUTABLE mysql_config
-    /usr/bin/
-    /usr/local/bin
-    ${MYSQL_DIR}/bin $ENV{MYSQL_DIR}/bin
-  )
+  # Split into two find_program invocations to avoid user-specified
+  # mariadb_config being overridden by system mysql_config.
+  find_program(MYSQL_CONFIG_EXECUTABLE NAMES mysql_config mariadb_config
+    PATH_SUFFIXES bin
+    NO_DEFAULT_PATH
+    HINTS ${MYSQL_DIR} $ENV{MYSQL_DIR}
```

Comment:
I would like to encourage users to use `CMAKE_PREFIX_PATH` more, which is the better way of adding extra search paths. Does this not work without these hints?

Reply:
Setting `CMAKE_PROGRAM_PATH` works in the case where the `XXX_config` program is available. I was not able to get `CMAKE_PREFIX_PATH` to work either with `XXX_config` or in the fallback case where one looks for the include and library areas separately. There are issues with respect to the alternative names for the config program, and it is unclear what the issue is (`PATH_SUFFIXES`?) in the fallback case.

Reply:
What directory are you adding to `CMAKE_PREFIX_PATH` and where is your `*_config`?

Reply:
`$(spack location -i mariadb-c-client)`, with `mariadb_config` being found in `bin/` directly beneath.